src/PublicApiAnalyzers/Core/CodeFixes/DeclarePublicApiFix.cs

```diff
@@ -295,10 +295,14 @@ protected override async Task<Solution> GetChangedSolutionAsync(CancellationToke
                     newSolution = newSolution.WithAdditionalDocumentText(pair.Key, pair.Value);
                 }
 
+                using var uniqueProjectPaths = PooledHashSet<string>.GetInstance();
```

Comment:
```suggestion
                // NOTE: We need to avoid creating duplicate files for multi-tfm projects. See https://github.com/dotnet/roslyn-analyzers/issues/3952.
                using var uniqueProjectPaths = PooledHashSet<string>.GetInstance();
```